### PR TITLE
Fix Android boot splash and gradle build issue

### LINKED ISSUE #99221: Image loading is broken in editor
### Tested versions

- Reproducible in master: v4.4.dev.custom_build [76fa7b291]
- Not reproducible in v4.4.dev4

### System information

Windows 11

### Issue description

I am cannot load images in editor.
I tried setting adaptive icons in android export and boot splash in project setting, both failed with similar error: `load_image: Error opening file 'uid://bjaelxrg45jgj'`

### Steps to reproduce

- Go to project settings
- Set a boot splash image (clear it if already set and select again)
- Run your project and see error in debugger.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by syntaxerror247**
This pull request is built upon PR https://github.com/godotengine/godot/pull/99286 and that needs to be merged first.

I have resolved the Android Gradle build export issue introduced in that PR, as well as fixed the boot splash problem on Android.

Fixes https://github.com/godotengine/godot/issues/99221

### CODE REVIEW

File: platform/android/export/export_plugin.cpp [hunk 1]
```diff
@@ -790,9 +790,10 @@ Error EditorExportPlatformAndroid::save_apk_so(void *p_userdata, const SharedObj
 
 Error EditorExportPlatformAndroid::save_apk_file(void *p_userdata, const String &p_path, const Vector<uint8_t> &p_data, int p_file, int p_total, const Vector<String> &p_enc_in_filters, const Vector<String> &p_enc_ex_filters, const Vector<uint8_t> &p_key, uint64_t p_seed) {
 	APKExportData *ed = static_cast<APKExportData *>(p_userdata);
-	String dst_path = p_path.replace_first("res://", "assets/");
+	String path = ResourceUID::ensure_path(p_path);
```
Comment (KoBeWi) [inline platform/android/export/export_plugin.cpp:793]:
```suggestion
	const String path = ResourceUID::ensure_path(p_path);
```
Same below.
`dst_path`  could be const too.

**Review by KoBeWi — APPROVED** [review]
I didn't test it, but it's similar to #99286

**Review by m4gr3d — APPROVED** [review]
Looks good!

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**